src/background/adblocker.js

```diff
@@ -167,7 +168,29 @@ export const setup = asyncSetup([
  * Cosmetics injection
  */
 
-async function injectScriptlets(scripts, tabId, frameId) {
+const contentScripts = (() => {
```

Comment:
`contentScripts` is not persistent, and with every reload of the background process, the map will be empty. The same hostname can be injected many times.

Reply:
content scripts registered by the `browser.contentScript` are also non persisted and live only as long as the background process. So this is correct

Reply:
Ok, I checked.

>Content scripts are also unregistered automatically when the page that created them is destroyed. For example, if they are registered from the background page they will be unregistered automatically when the background page is destroyed, and if they are registered from a sidebar or a popup, they will be unregistered automatically when the sidebar or popup is closed.